teps/0044-decouple-task-composition-from-scheduling.md

```diff
@@ -710,98 +834,109 @@ Things Pipelines can do that Tasks can't:
-     name: git-clone # uses our foobar above
- - name: run-unit-tests
-   runAfter: get-source
-   steps:
-  - name: unit-test
-    image: docker.io/library/golang:$(params.version)
-    script: <script here>
-   workspaces:
-   - name: source-code
-     workspcae: source-code
-   - name: test-results
-     workspace: test-results
- - name: upload-results
-   runAfter: run-unit-tests
-   foobarRef:
-     name: gcs-upload
-   params:
-   - name: location
-     value: gs://my-test-results-bucket/testrun-$(taskRun.name)
+ - name: source-code
+ - name: test-results
+ processes: # processes could be steps, processRefs, or processes themselves
+ - name: get-test-upload # this process is itself a process (a process in a process)
+   processes:
+   - name: get-source
+     workspaces:
+     - name: source-code
+       workspace: source-code
+     processRef: # this process is a ref
+       name: git-clone # uses our process above
+   - name: run-unit-tests
+     runAfter: get-source
+     steps: # this process is steps
+    - name: unit-test
+      image: docker.io/library/golang:$(params.go-version)
+      script: <script here>
+     workspaces:
+     - name: source-code
+       workspcae: source-code
+     - name: test-results
+       workspace: test-results
+   - name: upload-results
+     runAfter: run-unit-tests
+     processRef:
+       name: gcs-upload
+     params:
+     - name: location
+       value: gs://my-test-results-bucket/testrun-$(taskRun.name)
+     workspaces:
+     - name: data
+       workspace: test-results
+ - name: deploy
+   runAfter: [get-test-upload]
+   when:
+   - input: "$(params.git-branch)"
+     operator: in
+     values: ["main"]
+   processRef:
+     name: build-deploy
    workspaces:
-   - name: data
-     workspace: test-results
+    - name: source-code
+      workspcae: source-code
 finally:
 - name: update-slack
   params:
   - name: message
     value: "Tests completed with $(tasks.run-unit-tests.status) status"
 ```
 
-We will need to add something to indicate how to schedule Foobars now that we won't have the convenience of drawing the
-line around Tasks; we could combine this idea with one of the others in this proposal.
+Processes contain Processes, which can be defined as steps or reference to other Processes.
+
+We will need to add something to indicate how to schedule Processes now that we won't have the convenience of drawing the
+line around Tasks; we could combine this idea with one of the others in this proposal - likely we'd have to introduce
+new syntax to express scheduling, and/or we could have similar defaults (e.g. steps execute together in one pod).
 
 Pros:
 * Maybe the distinction between Tasks and Pipelines has just been making things harder for us
 * Maybe this is a natural progression of the "embedding" we already allow in pipelines?
 * We could experiment with this completely independently of changing our existing CRDs (as long as we don't want
```

Comment:
Note : we can experiment this *completely* independently of Tekton itself 😅. The idea though, is that people are adopting Tekton as we speak, they are learning Pipeline, Task, … do we want to through them away and say "use this new thing" ?